libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -249,16 +258,33 @@ - (SFRestRequest *)requestForCreateWithObjectType:(NSString *)objectType
 - (SFRestRequest *)requestForUpdateWithObjectType:(NSString *)objectType
                                          objectId:(NSString *)objectId
                                            fields:(NSDictionary *)fields {
+    return [self requestForUpdateWithObjectType:objectType objectId:objectId fields:fields ifUnmodifiedSinceDate:nil];
+}
+
+- (SFRestRequest *)requestForUpdateWithObjectType:(NSString *)objectType
+                                         objectId:(NSString *)objectId
+                                           fields:(NSDictionary *)fields
+                            ifUnmodifiedSinceDate:(NSDate *) ifUnmodifiedSinceDate {
+
     NSString *path = [NSString stringWithFormat:@"/%@/sobjects/%@/%@", self.apiVersion, objectType, objectId];
-    return [SFRestRequest requestWithMethod:SFRestMethodPATCH path:path queryParams:fields];
+    SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodPATCH path:path queryParams:fields];
+    [request setHeaderValue:[SFRestAPI getHttpStringFomFromDate:ifUnmodifiedSinceDate] forHeaderName:kSFRestIfUnmodifiedSince];
```

Comment:
Should we be setting a header value for a `nil` result?  Is there a reason for `ifUnmodifiedSinceDate` to be `nullable`?